core/src/toga/style/pack.py

```diff
@@ -36,196 +35,273 @@
-        }
-
-    def __get__(self, obj, objtype=None):
-        if obj is None:
-            return self
-
-        self.warn_if_deprecated()
-
-        if hasattr(obj, f"_{self.other}"):
-            # If the other property is set, attempt to translate.
-            for condition, value in self.derive.items():
-                if condition.match(obj, main_name=self.other):
-                    return value
-
-        # If the other property isn't set (or no condition is valid), access this
-        # property as usual.
-        return super().__get__(obj)
-
-    def __set__(self, obj, value):
-        if value is self:
-            # This happens during autogenerated dataclass __init__ when no value is
-            # supplied.
-            return
-
-        self.warn_if_deprecated()
-
-        # Delete the other property when setting this one.
-        try:
-            delattr(obj, f"_{self.other}")
-        except AttributeError:
-            pass
-        super().__set__(obj, value)
-
-    def __delete__(self, obj):
-        self.warn_if_deprecated()
-
-        # Delete the other property too.
-        try:
-            delattr(obj, f"_{self.other}")
-        except AttributeError:
-            pass
-        super().__delete__(obj)
-
-    def is_set_on(self, obj):
-        self.warn_if_deprecated()
-
-        # Counts as set if *either* of the two properties is set.
-        return super().is_set_on(obj) or hasattr(obj, f"_{self.other}")
-
-    def warn_if_deprecated(self):
-        if self.name == "alignment":
-            warnings.warn(
-                "Pack.alignment is deprecated. Use Pack.align_items instead.",
-                DeprecationWarning,
-                stacklevel=3,
-            )
-
-
-######################################################################
-# End backwards compatibility
-######################################################################
-
 
 @dataclass(kw_only=True, repr=False)
-class Pack(BaseStyle):
+class Pack(PackMechanicsBase):
     _doc_link = "[style properties](/reference/style/pack)"
 
-    class Box(BaseBox):
-        pass
-
-    class IntrinsicSize(BaseIntrinsicSize):
-        pass
+    display: str = validated_property(PACK, NONE, initial=PACK)
+    """**Value:** `"pack"` or `"none"`
 
-    _depth = -1
+    **Initial value:** `"pack"`
 
-    display: str = validated_property(PACK, NONE, initial=PACK)
+    Used to define how to display the widget. A value of `"pack"` will apply the pack
+    layout algorithm to this node and its descendants. A value of `"none"` removes the
+    widget from the layout entirely. Space will be allocated for the widget as if it
+    were there, but the widget itself will not be visible.
+    """
     visibility: str = validated_property(VISIBLE, HIDDEN, initial=VISIBLE)
+    """**Value:** `"hidden"` or `"visible"`
+
+    **Initial value:** `"visible"`
+
+    Used to define whether the widget should be drawn. A value of `"visible"` means the
```

Comment:
After looking into it, I see that this is really the only option for documenting this information. That said, I think for consistency, it should be rearranged a bit. Here is what I'd like to see. 
```suggestion
    """
    Used to define whether the widget should be drawn. 
    
    **Value:** `"hidden"` or `"visible"`

    **Initial value:** `"visible"`

    A value of `"visible"` means the
```